sites/main-site/src/content/blog/2024/maintainers-minutes-2024-10-25.mdx

```diff
@@ -0,0 +1,167 @@
+
+This proposal initially brought up a lot of consternation with most of the other maintainers present. The primary issue was the potential impact on usability, specifically regarding the ease with which users can locate customized parameters within each module.
+Currently, we can direct users to a single file within a single directory, simplifying their search.
+
+Splitting configurations into potentially hundreds of files could make it significantly harder for users to find what they need. It's also unclear if another file has an overriding setting for a module, for example that's defined in the subworkflow folder rather than the module folder. Additionally, this approach isn’t currently supported by Nextflow’s native capabilities, which would allow modules to automatically detect and load nextflow.config files alongside main.nf. As a result, a substantial amount of manual work would be needed to manually include each module’s config.
+
+A lengthy discussion ensued, weighing user readability against developer efficiency and which should take priority. For instance, efficiency improvements could be made at the level of nf-core/tools or GitHub CI configurations, where optimizations might achieve similar goals.
+Alternatively, waiting for native Nextflow support could eventually reduce the workload on developers.
+
+The sarek developers proposed a middle ground that has worked well in sarek: triggering tests on smaller chunks while maintaining findability. Their approach involves naming config files after each process and storing all configs within a `conf/modules/` folder. This setup makes configurations easy to locate and include while still achieving modularity. see sarek's module configs organization [here](https://github.com/nf-core/sarek/tree/master/conf/modules)
+
+A general agreement was that this new system is not yet palatable to enough maintainers to propose push to the template, however that `nf-core/rnaseq` and `nf-core/methylseq` can continue to refine this approach based on what `nf-core/sarek` developers have proposed.
+
+Developers agreed to revisit and review this approach again in a couple of months.
+
+## Modification of modules specifications regarding channels
+
+Finally, we debated pros- and cons- of different ways of structuring input channels in modules.
+
+In the initial design of DSL2 nf-core modules, it was decided to require one input channel per file type, and not support 'mega-tuples', due to readability and portability.
+
+For example, this was found to be less readable:
+
+```nextflow
+input:
+tuple value(meta), path(bam), path(bai), path(fasta), path(fai), path(dict)
+```
+
+Compared to:
+
+```nextflow
+input:
+tuple value(meta), path(bam)
+tuple value(meta), path(bai)
+tuple value(meta), path(fasta)
+tuple value(meta), path(fai)
+tuple value(meta), path(dict)
+```
+
+Where ensuring each file was associated with it's companion could be facilitated with (relatively) standardised `multiMap()` calls.
```

Comment:
We should have an example here:

````suggestion
Where ensuring each file was associated with it's companion could be facilitated with (relatively) standardised `multiMap()` calls.

```nextflow
combined_channel.multiMap { meta, bam, bai, fasta, fai, dict ->
    bam: tuple( meta, bam )
    bai: tuple( meta, bai )
    fasta: tuple ( meta, fasta )
    fai: tuple ( meta, fai )
    dict: tuple ( meta, dict )
}
.set { process_in }
PROCESS (
    process_in.bam,
    process_in.bai,
    process_in.fasta,
    process_in.fai,
    process_in.dict,
)
```

:::info
`multiMap` should only be used in this way when directly providing process inputs. It's not a general method of synchronising channel inputs.
:::
````